docs/oathkeeper/guides/websockets.mdx

```diff
@@ -0,0 +1,359 @@
+- 4434 is an admin API of Ory Kratos
+- 4455 is a port of the User Interface implemented by self-service-ui
+- 8080 is a port of Ory Oathkeeper
+
+The rest ports or services will be available only inside the internal network.
+
+## Configuring Ory Oathkeeper and Ory Kratos
+
+Let's create a `kratos` folder and fetch configuration files
+
+<CodeBlock className="language-shell">{`mkdir kratos
+wget https://raw.githubusercontent.com/ory/kratos/${useLatestRelease('kratos')}/contrib/quickstart/kratos/email-password/identity.schema.json -O kratos/identity.schema.json
+wget https://raw.githubusercontent.com/ory/kratos/${useLatestRelease('kratos')}/contrib/quickstart/kratos/email-password/kratos.yml -O kratos/kratos.yml`}</CodeBlock>
+
+As a next step, create `oathkeeper` folder and `oathkeeper/oathkeeper.yml` with the following content
+
+```yaml
+log:
+  level: debug
+  format: json
+
+serve:
+  proxy:
+    cors:
+      enabled: true
+      allowed_origins:
+        - http://127.0.0.1:8080
+      allowed_methods:
+        - POST
+        - GET
+        - PUT
+        - PATCH
+        - DELETE
+      allowed_headers:
+        - Authorization
+        - Content-Type
+      exposed_headers:
+        - Content-Type
+      allow_credentials: true
+      debug: true
+
+errors:
+  fallback:
+    - json
+
+  handlers:
+    redirect:
+      enabled: true
+      config:
+        to: http://127.0.0.1:4455/login
+        when:
+          -
+            error:
+              - unauthorized
+              - forbidden
+            request:
+              header:
+                accept:
+                  - text/html
+    json:
+      enabled: true
+      config:
+        verbose: true
+
+access_rules:
+  matching_strategy: glob
+  repositories:
+    - file:///etc/config/oathkeeper/access-rules.yml
+
+authenticators:
+  anonymous:
+    enabled: true
+    config:
+      subject: guest
+
+  cookie_session:
+    enabled: true
+    config:
+      check_session_url: http://kratos:4433/sessions/whoami
+      preserve_path: true
+      extra_from: "@this"
+      subject_from: "identity.id"
+      only:
+        - ory_kratos_session
+
+  noop:
+    enabled: true
+
+authorizers:
+  allow:
+    enabled: true
+
+mutators:
+  noop:
+    enabled: true
+```
+
+and `oathkeeper/access-rules.yml` with the following content
+
+```yaml
+-
+  id: "ws:protected"
+  upstream:
+    preserve_host: true
+    url: "http://ws:8080"
+  match:
+    url: "http://127.0.0.1:8080/<**>"
+    methods:
+      - GET
+      - POST
+  authenticators:
+    - handler: cookie_session
+  mutators:
+    - handler: noop
+  authorizer:
+    handler: allow
+  errors:
+    - handler: redirect
+      config:
+        to: http://127.0.0.1:4455/login
+```
+
+This configuration of oathkeeper uses cookie authenticator against Ory Kratos and proxies only authenticated requests to `http://ws:8080` upstream. The ws hostname is a hostname of the websocket service that we'll create right now.
```

Comment:
Just add here that `ws:8080` is resolved through the docker network. If you aren't deploying your application within docker, this would just be your localhost IP.

Otherwise we will get people confused that do not know or understand docker or its networks :sweat_smile: